src/shared/rpc/methods/validation.ts

```diff
@@ -0,0 +1,60 @@
+import { PaymentTypes } from '@btckit/types';
+import * as btc from '@scure/btc-signer';
+import * as yup from 'yup';
+
+import {
+  formatValidationErrors,
+  getRpcParamErrors,
+  testIsNumberOrArrayOfNumbers,
+  validateRpcParams,
+} from './validation.utils';
+
+// TODO: Import Bip322MessageTypes from btckit when updated
+type SupportedBip322MessageTypes = 'bip322';
+
+const accountSchema = yup.number().integer();
+
+export const rpcSignPsbtParamsSchema = yup.object().shape({
+  allowedSighash: yup.array().of(yup.mixed().oneOf(Object.values(btc.SignatureHash))),
+  hex: yup.string().required(),
+  signAtIndex: yup.mixed<number | number[]>().test(testIsNumberOrArrayOfNumbers),
+  account: accountSchema,
+});
+
+const rpcSendTransferParamsSchema = yup.object().shape({
+  account: accountSchema,
+  address: yup.string().required(),
+  amount: yup.string().required(),
+});
+
+const rpcSignMessageParamsSchema = yup.object().shape({
+  type: yup.string<SupportedBip322MessageTypes>(),
+  account: accountSchema,
+  message: yup.string().required(),
+  paymentType: yup.string<PaymentTypes>().required(),
+});
+
+// TODO: Import param types from btckit when updated
+export function validateRpcSignPsbtParams(obj: unknown) {
+  return validateRpcParams(obj, rpcSignPsbtParamsSchema);
+}
+
+export function getRpcSignPsbtParamErrors(obj: unknown) {
+  return formatValidationErrors(getRpcParamErrors(obj, rpcSignPsbtParamsSchema));
+}
+
+export function validateRpcSendTransferParams(obj: unknown) {
+  return validateRpcParams(obj, rpcSendTransferParamsSchema);
+}
+
+export function getRpcSendTransferParamErrors(obj: unknown) {
+  return formatValidationErrors(getRpcParamErrors(obj, rpcSendTransferParamsSchema));
+}
+
+export function validateRpcSignMessageParams(obj: unknown) {
+  return validateRpcParams(obj, rpcSignMessageParamsSchema);
+}
+
+export function getRpcSignMessageParamErrors(obj: unknown) {
+  return formatValidationErrors(getRpcParamErrors(obj, rpcSignMessageParamsSchema));
+}
```

Comment:
Can we keep colocated to the methods themselves? I've tried to organise them `src/shared/rpc/methods/{file-per-method}` this goes against that pattern.

Reply:
Ok, yep, np.